File: payroll-api/src/main/java/com/company/payroll/mapper/FileAttachmentMapper.java
package com.company.payroll.mapper;

import java.util.List;

import org.springframework.stereotype.Repository;

import com.company.payroll.model.FileAttachment;

@Repository
public interface FileAttachmentMapper {
	int deleteByPrimaryKey(Integer fId);
	int insert(FileAttachment row);
	int insertSelective(FileAttachment row);
	List<FileAttachment> selectListByStaffId(Integer staffId);
	FileAttachment selectByPrimaryKey(Integer fId);
	int updateByPrimaryKeySelective(FileAttachment row);
	int updateByPrimaryKey(FileAttachment row);

	/**
	 * @deprecated will be removed after FileAttachmentMapper has success implemented.
	 * <p>Please use {@link #selectListByStaffId(Integer)} instead.}</p>
	 * @param mId
	 * @return
	 */
	List<FileAttachment> selectListByMId(Integer mId);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/FileAttachmentMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.FileAttachment
import org.springframework.stereotype.Repository

@Repository
interface FileAttachmentMapper {
  fun deleteByPrimaryKey(fId: Int): Int

  fun insert(row: FileAttachment): Int

  fun insertSelective(row: FileAttachment): Int

  fun selectByPrimaryKey(fId: Int): FileAttachment

  fun selectListByStaffId(staffId: Int): List<FileAttachment>

  fun updateByPrimaryKeySelective(row: FileAttachment): Int

  fun updateByPrimaryKey(row: FileAttachment): Int
}